ledger/src/blockstore.rs

```diff
@@ -2898,18 +2901,33 @@ impl Blockstore {
         keys_with_writable: impl Iterator<Item = (&'a Pubkey, bool)>,
         status: TransactionStatusMeta,
         transaction_index: usize,
+        db_write_batch: Option<&mut WriteBatch<'_>>,
```

Comment:
I'm thinking it might be better to split this into two APIs. One for the existing behavior and one for the new behavior (called something like `transaction_status_add_to_batch`)

Reply:
Agreed. Let's create an overload. Mixing single and batch is complicated.